schema/shared/src/main/scala/zio/blocks/schema/SchemaError.scala

```diff
@@ -72,4 +72,48 @@ object SchemaError {
   case class UnknownCase(source: DynamicOptic, caseName: String) extends Single {
     override def message: String = s"Unknown case '$caseName' at: $source"
   }
+
+  // Patch-specific error types
+  case class FieldNotFoundError(source: DynamicOptic, fieldName: String, availableFields: List[String]) extends Single {
+    override def message: String = s"Field '$fieldName' not found at: $source. Available fields: ${availableFields.mkString(", ")}"
+  }
+
+  case class TypeMismatchError(source: DynamicOptic, expected: String, actual: String) extends Single {
+    override def message: String = s"Type mismatch at: $source. Expected $expected, got $actual"
+  }
+
+  case class CaseMismatchError(source: DynamicOptic, expected: String, actual: String) extends Single {
+    override def message: String = s"Case mismatch at: $source. Expected $expected, got $actual"
+  }
+
+  case class IndexOutOfBoundsError(source: DynamicOptic, index: Int, size: Int) extends Single {
+    override def message: String = s"Index $index out of bounds (size: $size) at: $source"
+  }
+
+  case class KeyNotFoundError(source: DynamicOptic, key: String) extends Single {
+    override def message: String = s"Key '$key' not found at: $source"
+  }
+
+  case class KeyAlreadyExistsError(source: DynamicOptic, key: String) extends Single {
+    override def message: String = s"Key '$key' already exists at: $source"
+  }
+
+  // Factory methods for patch operations (use root optic by default)
+  def FieldNotFound(fieldName: String, availableFields: List[String]): SchemaError =
```

Comment:
These are not storing full optics to the field but just field names, which is mistaken. Also tests should have caught these errors which means the tests need to be checking for full path information in the errors.